README.md

```diff
@@ -39,10 +39,10 @@ jobs:
         uses: DataDog/ensure-ci-success@v1
 ```
 
-It's a good practice to set this job as a final job in your pipeline using `needs` parameter, to limit useless CPU time.
-
 The final step is to make this job a requirements for merges using branch protection rules.
 
+It's a good practice to set this job as a final job in your pipeline using `needs` parameter, to limit useless CPU time. Though, if you do so, any failure will prevent this last job to run, and Github Pr engine does not account it as failed. So do not forget to add a parameter `if: '!cancelled()'`, it will force the job to run, unless it's explicitly cancelled.
```

Comment:
Typically, use `if: '!cancelled()'` instead of `always()`. But this job should be `always()`. 

Because it job is only going to be executed if upstream jobs are NOT cancelled. When the upstream jobs are cancelled, the job is marked as `skipped`. Being `skipped` means it becomes irrelevant, hence allowing the merged.

Reply:
Confirmed that missing the `if` statement does not achieve what we want : 

<img width="362" alt="image" src="https://github.com/user-attachments/assets/efdd6c57-6e16-40da-947c-eccbeab149ff" />

Reply:
if a previous job is cancelled, then it's fine, the PR is blocked (i've used `if: '!cancelled()'` here)

<img width="350" alt="image" src="https://github.com/user-attachments/assets/b42b8813-dca4-48ca-a63d-547f8486000c" />

Reply:
And if the job itself is cancelled, it's correctly blocked : 

<img width="932" alt="image" src="https://github.com/user-attachments/assets/3fcf6c7c-6bb9-4b9f-8da5-5e277fdc9fce" />